### PR TITLE
feat: new construct textract to comprehend medical

### PR DESCRIPTION
**Pull request by svurputoor**
Fixes #

### CODE REVIEW

File: lambda/textract_comprehend_medical/env.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "Parameters": {
+        "HEALTHLAKE_ENDPOINT": "https://healthlake.us-east-2.amazonaws.com/datastore/83c10afe8566667ac2489c8d989b2c14/r4/",
```
Comment (MichaelWalker-git) [inline lambda/textract_comprehend_medical/env.json:3]: Are these something that any external customer can ingest?

File: src/textractComprehendMedical.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+ */
+export class TextractComprehendMedical extends sfn.StateMachineFragment {
+  public readonly startState: sfn.State;
+  public readonly endStates: sfn.INextable[];
+  public readonly textractComprehendMedicalFunction: lambda.IFunction;
+
+  constructor(parent: Construct, id: string, props: TextractComprehendMedicalProps) {
+    super(parent, id);
+
+    var lambdaMemoryMB = props.lambdaMemoryMB === undefined ? 1024 : props.lambdaMemoryMB;
```
Comment (MichaelWalker-git) [inline src/textractComprehendMedical.ts:52]: Replace var with const or let.

https://airbnb.io/javascript/#variables

File: src/textractComprehendMedical.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    var cmJobType = props.comprehendMedicalJobType === undefined ? 'ICD10' : props.comprehendMedicalJobType;
+
+    const comprehendMedicalRole = new iam.Role(this, 'RoleComprehendMedical', {
+      assumedBy: new iam.ServicePrincipal('comprehendmedical.amazonaws.com'),
+    });
+    comprehendMedicalRole.addToPolicy(
+      new iam.PolicyStatement({
+        actions: ['s3:GetObject', 's3:ListBucket', 's3:PutObject'],
+        effect: iam.Effect.ALLOW,
+        resources: ['*'],
```
Comment (MichaelWalker-git) [inline src/textractComprehendMedical.ts:65]: Any chance we can narrow this down?

File: src/textractComprehendMedical.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+              path.join(`arn:aws:s3:::${props.s3InputBucket}`, s3InputPrefix, '/'),
+              path.join(`arn:aws:s3:::${props.s3InputBucket}`, s3InputPrefix, '/*'),
+            ],
+          }),
+        );
+      }
+      this.textractComprehendMedicalFunction.addToRolePolicy(
+        new iam.PolicyStatement({
+          actions: ['comprehendmedical:Start*', 'iam:PassRole'],
+          resources: ['*'],
```
Comment (MichaelWalker-git) [inline src/textractComprehendMedical.ts:109]: Same as above.

**Review by MichaelWalker-git — CHANGES_REQUESTED** [review]
Overall, this looks good. 

I would also recommend that we add changes to the README or more documentation on how to use it.

File: lambda/textract_comprehend_medical/env.json (hunk 1 above)
Comment (svurputoor) [reply lambda/textract_comprehend_medical/env.json:3]: This is a local test file

File: src/textractComprehendMedical.ts (hunk 3 above)
Comment (svurputoor) [reply src/textractComprehendMedical.ts:65]: I changed the code to cover both situations 1/ when I have input bucket and 2/ when I don't have input bucket

File: lambda/textract_comprehend_medical/app/fhir_doc_assembler.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import base64
+
+
+class FhirDocAssembler():
+    def __init__(self):
+        self.attachments = []
+
+    # def add_attachment(self, raw_text):
```
[the remark this answers is not in the document]
Comment (svurputoor) [reply lambda/textract_comprehend_medical/app/fhir_doc_assembler.py:8]: deleted

File: lambda/textract_comprehend_medical/app/main.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+
+# Broken out into separate Lambda function in case there are slow-downs in textract,
+# the document is very large, or in general if the time needed for Textract exceeds
+# the 15-minute limit of Lambda.  Also, no need to leave a Lambda function running
+# while it actively polls Textract when Textract is perfectly happy notifying us on
+# SNS when it's ready for us to come back :-)
+
+def handler(event, context):
+    logger.debug(event)
```
[the remark this answers is not in the document]
Comment (svurputoor) [reply lambda/textract_comprehend_medical/app/main.py:29]: deleted

File: src/textractComprehendMedical.ts (hunk 2 above)
Comment (svurputoor) [reply src/textractComprehendMedical.ts:52]: replaced with const

File: src/textractComprehendMedical.ts (hunk 4 above)
Comment (svurputoor) [reply src/textractComprehendMedical.ts:109]: https://docs.aws.amazon.com/comprehend-medical/latest/dev/security-iam-permissions.html#access-policy-aws-managed-policies-med
we are following this

**Review by MichaelWalker-git — APPROVED** [review]
LGTM

**Pull request merged**